snow/engine/snowman/transitive.go

```diff
@@ -259,30 +260,43 @@ func (t *Transitive) PushQuery(ctx context.Context, nodeID ids.NodeID, requestID
 	return t.buildBlocks(ctx)
 }
 
-func (t *Transitive) Chits(ctx context.Context, nodeID ids.NodeID, requestID uint32, blkID ids.ID, acceptedID ids.ID) error {
+func (t *Transitive) Chits(ctx context.Context, nodeID ids.NodeID, requestID uint32, preferredID ids.ID, preferredIDAtHeight ids.ID, acceptedID ids.ID) error {
 	t.acceptedFrontiers.SetLastAccepted(nodeID, acceptedID)
 
 	t.Ctx.Log.Verbo("called Chits for the block",
-		zap.Stringer("blkID", blkID),
 		zap.Stringer("nodeID", nodeID),
-		zap.Uint32("requestID", requestID))
+		zap.Uint32("requestID", requestID),
+		zap.Stringer("preferredID", preferredID),
+		zap.Stringer("preferredIDAtHeight", preferredIDAtHeight),
+		zap.Stringer("acceptedID", acceptedID),
+	)
 
-	added, err := t.issueFromByID(ctx, nodeID, blkID)
+	addedPreferred, err := t.issueFromByID(ctx, nodeID, preferredID)
 	if err != nil {
 		return err
 	}
 
-	// Will record chits once [blkID] has been issued into consensus
+	addedPreferredIDAtHeight, err := t.issueFromByID(ctx, nodeID, preferredIDAtHeight)
+	if err != nil {
+		return err
+	}
+
+	// Will record chits once [preferredID] and [preferredIDAtHeight] have been
+	// issued into consensus
```

Comment:
So, even if we failed here https://github.com/ava-labs/avalanchego/pull/2102/files#diff-1b154e2d9b5b0909a66199417df6ff79ee9a18e83075b39ef2acf3634558342eR724-R736 (non-populated preferred id at height set to preferred id), this is safe? can we doc regarding v1.11 upgrade?

Reply:
The invariants for a correct node is that `preferredID` transitively references both `preferredIDAtHeight ` and `acceptedID`. So it would be safe to default `preferredIDAtHeight` to either `preferredID` or `acceptedID` in the handler